CHERI-UPDATING.md

```diff
@@ -0,0 +1,86 @@
+# Updating Information for CheriBSD users.
+
+This file contains information about updating CheriBSD.  It supplements
+the information in the [UPDATING] file with CheriBSD specific
+information.  In particular it explains ABI flag days, our branching structure,
+and our stragety from merging from upstream FreeBSD.
+
+## Flag days and major changes
+
+Generally speaking, flag days require a complete rebuild of CheriBSD.
+When building with [cheribuild] this is usually handled automatically,
+but if necessicary the `--clean` flag can be used.  When building
+directly, avoid the `-DNO_CLEAN` make option.
+
+### Flag days on [master]
+
+| Date       | Commit      | Description | Required action |
+| ---------- | ----------- | --- | --- |
+| 2020-03-06 | [6ce214d1e] | ELF auxargs flags altered | Clean rebuild |
+
+### Flag days on [dev]
+
+| Date       | Commit      | Description | Required action |
+| ---------- | ----------- | --- | --- |
+| 2020-03-21 | [7e76d8f71] | C/C++ ABI changes | Update to [LLVM b7f5c847dc] and a clean rebuild.|
```

Comment:
How will this appear when we merge to master. Will we give the original date/commit, or that of the merge? We need to be communicating both clearly.

Reply:
My thought is that we'll give the date of the merge and the merge commit (maybe with the merged commit in the description.)  Awkwardly, that means a double merge where we commit with an XXXX in place of the Commit and then commit the real entry.

Reply:
Or you just push the update to this file separately as a followup change.